### PR TITLE
feat(policies) Add tests for Throw + Silent policy

### LINKED ISSUE #111: feat(policies) Add tests for Throw + Silent policy
### Describe Your Feature

This issue focuses on adding comprehensive unit tests for the Throw and Silent logging policies.

- Throw Policy: Any exception raised must be properly caught and verified against the expected graph exception type.

- Silent Policy: No logs should be written to either the log file or the console.

### Describe the Implementation Plan.

The goal is to ensure this policy configuration behaves exactly as intended.

Reference: Take a look at the existing test file for a similar configuration:
[EPShard_ThrowAndLog.cpp](https://github.com/SharonIV0x86/CinderPeak/blob/main/tests/unit/Policies/EPShard_ThrowAndLog.cpp)

Add a new test file called ``EPShard_ThrowAndSilent.cpp`` in the ``/tests/Policies`` folder.

### PR DESCRIPTION
**Pull request by prasadsarwar**
## Which issue does this PR close?
- Closes #111.

## Rationale for this change
This PR addresses issue #111 by adding unit tests for the Throw and Silent error handling policies in the CinderPeak project, ensuring comprehensive test coverage for all `PeakStatus` conditions. The tests verify that the Throw policy raises correct exceptions and document a bug in the Silent policy, which fails to suppress console output, to aid future improvements.

## What changes are included in this PR?
- Added `tests/unit/Policies/EPShard_ThrowAndSilent.cpp` with 12 unit tests:
  - 9 tests covering all `PeakStatus` conditions (NotFound, InvalidArgument, VertexAlreadyExists, InternalError, EdgeNotFound, VertexNotFound, Unimplemented, AlreadyExists, EdgeAlreadyExists).
  - Tests for handling multiple exceptions in sequence, repeated exception handling, and EdgeNotFound with a custom message.
  - Fixed a typo in the EdgeAlreadyExists test (`aeex` to `eaex`).
  - Adjusted output expectations to match current `PolicyHandler::handleException` behavior, which outputs "Set the error policy as throw" due to a bug in `PolicyConfiguration::Silent`.
- Noted the Silent policy bug and suggested a fix to move output to `Logger::log` to respect the Silent policy.

## Are these changes tested?
Yes, the PR includes 12 unit tests in `tests/unit/Policies/EPShard_ThrowAndSilent.cpp`, all of which pass (verified with `ctest -R ThrowAndSilent -V`). The tests cover:
- Correct exception throwing for each `PeakStatus` condition.
- Silent policy behavior (currently expecting output due to the bug).
- Edge cases like multiple and repeated exceptions, and custom error messages.

## Are there any user-facing changes?
No, these changes are limited to adding unit tests and do not affect user-facing functionality or public APIs.

### CODE REVIEW

**Comment by SharonIV0x86** [comment]
@prasadsarwar Your work is admirable.

But there is a small change that needs to be done.
Currently the policy class puts these two ``std::cout``
```cpp
      std::cout << "Set the error policy as ignore\n";
```
and
```cpp
      std::cout << "Set the error policy as throw\n";
```
 While developing the policy handler i added them only for debugging purpose and this will be removed soon as we cannot have random raw ``std::cout`` in production mode.

Instead of checking for a specific message in the ``stdout`` the tests should check that nothing was written to the ``stdout`` if the policies specify Throw + Silent. So the current approach can be changed to:

```cpp
  std::string output = testing::internal::GetCapturedStdout();
  EXPECT_TRUE(output.empty())  << "Expected no console output, but got: " << output;
```
So, if possible kindly remove those two ``std::cout`` statements as well. 

And let me know if you are contributing for GSSoC or Hacktoberfest or both so i can add labels to the PR.

Regards,
SharonIV0x86

**Comment by prasadsarwar** [comment]
Hey @SharonIV0x86 i have done the following changes
- Updated `src/PeakLogger.hpp`:
  - Modified `Logger::log` to return immediately for `Silent` policy, preventing console or file output.
  
- Updated `src/PolicyConfiguration.hpp`:
  - Fixed `PolicyHandler::handleException` to use correct `Logger::log` signature with loggingPolicy and logFilePath parameters.
  - Removed `std::cout` statements ("Set the error policy as ignore" and "Set the error policy as throw") to ensure all logging respects Silent policy
  
  I am contributing for Hacktoberfest 2025, would you please add the `Hacktoberfest` label.

File: tests/unit/Policies/EPShard_ThrowAndSilent.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+// 1. NotFound exception
+TEST_F(ThrowAndSilentPolicyTest, ThrowAndSilent_NotFound) {
+  testing::internal::CaptureStdout();
+  try {
+    policy.handleException(sc_notFound);
+  } catch (const PeakExceptions::NotFoundException &nfex) {
+    EXPECT_STREQ(nfex.what(), "Resource Not Found: Not Found");
+  }
+  std::string output = testing::internal::GetCapturedStdout();
```
Comment (SharonIV0x86) [inline tests/unit/Policies/EPShard_ThrowAndSilent.cpp]: In the similar way the ``PeakLogger`` also writes to ``std::cerr`` in some cases so, it would be great if tests also validate that nothing was written to the ``std::cerr`` stream as well.
Something similar can be implemented very easily with ``GetCapturedStderr()`` for all the test cases.

**Comment by prasadsarwar** [comment]
@SharonIV0x86 Thank you for the feedback! I've updated all 12 tests in `tests/unit/Policies/EPShard_ThrowAndSilent.cpp` to capture and validate `std::cerr` using `testing::internal::CaptureStderr()` and `EXPECT_TRUE(stderr_output.empty())`, with error messages changed to 'Expected no stderr output' for clarity. I kept `std::cerr` in `PeakLogger.hpp`’s `logToConsole` function since it’s suppressed by the Silent policy, as validated by the tests. I rebased the `fix-issue-111` branch to include the latest remote changes. All tests pass (verified with `ctest -R ThrowAndSilent -V`). Thank you for the `hacktoberfest` label for Hacktoberfest 2025! Please let me know if I should replace `std::cerr` with `std::cout` in `PeakLogger.hpp` for consistency or if any other changes are needed.

**Review by SharonIV0x86 — APPROVED** [review]
Your PR has been approved and it will be merged in sometime. Till then you can work on any other issue if you'd like.

**Comment by prasadsarwar** [comment]
@SharonIV0x86 Thank you so much for approving PR #112! I’m grateful for your feedback and the `hacktoberfest` label for Hacktoberfest 2025. I’ll wait for the merge and am excited to contribute more. Thanks again!

**Comment by SharonIV0x86** [comment]
@prasadsarwar Thank you for contributing, your PR has been merged. Kindly ⭐ the repository as it shows appreciation to the project maintainers.
